### PR TITLE
Create a sync-repos action.

### PR DESCRIPTION
**Pull request by chandlerc**
This will allow us to automatically create and maintain specific repos
based on the main repository here, pulling key files like the license
and other infrastructure and pushing them systematically to a narrow
repo. Things like editor plugins that are best packaged and installed
from a separate repos can still be developed in a central place, even
potentially sharing common things like grammars where useful.

Currently this will maintain a Vim plugin repository out of the
`utils/vim` directory, but can be easily expanded for other systems.

### CODE REVIEW

File: scripts/sync_repos.sh
Comment (jonmeow) [inline scripts/sync_repos.sh:15]: This implementation feels pretty difficult to understand right now, and I think the multiline strings are adding to my difficult. What would you think of using Python with an eye on long-term maintainability?

File: scripts/sync_repos.sh
Comment (jonmeow) [inline scripts/sync_repos.sh]: I'm a bit confused by this; what's the goal of printing this? Won't `rsync -av` already print a lot?

This makes me wonder though: what does the output of this script look like? Is it easy to read? It seems like it may be a bunch of text.

**Comment by jonmeow** [comment]
The point that sync-repos is running on this PR is also an issue; it shouldn't be running on pending PRs.

**Comment by jonmeow** [comment]
Stepping back, a cron approach rather than a per-commit approach may make more sense. It does make me wonder though if writing your own is really the best choice.

File: .github/workflows/sync_repos.yaml
Comment (jonmeow) [inline .github/workflows/sync_repos.yaml]: IIUC, `sync-repos` needs to be removed here (this blocks merging)

**Comment by chandlerc** [comment]
We discussed some of the meta things in person, but ot re-cap...

> The point that sync-repos is running on this PR is also an issue; it shouldn't be running on pending PRs.

It's not running on arbitrary PRs, it's *exclusively* this PR because I created this in a branch of the main repo so that I could test it while developing it. =] Prior to merge, this branch will be removed from the action so it only runs on trunk from then on.


> Stepping back, a cron approach rather than a per-commit approach may make more sense. It does make me wonder though if writing your own is really the best choice.

The goal is specifically to mirror commits directly. That seems much simpler. The way this should work is that once landed this will only run when we actually merge into trunk, and then it will mirror that commit to the other repos.

File: .github/workflows/sync_repos.yaml
Comment (chandlerc) [reply .github/workflows/sync_repos.yaml]: Yes, this was just while developing. Now that its working and I think ready for merge, removed.

File: scripts/sync_repos.sh
Comment (chandlerc) [reply scripts/sync_repos.sh]: It'll be somewhat verbose, but I think that's a better tradeoff for a GitHub action to make it more debuggable when it goes wrong. This isn't *recursive* note, so it shouldn't be a huge amount.

And this is printing out *before* we sync things.

File: scripts/sync_repos.sh
Comment (chandlerc) [reply scripts/sync_repos.sh:15]: I'm not sure about the tradeoff here.

I think we're at >50% running commands, which I find much harder to read in Python. I think there are only three things that are more fancy that I'd like in shell here -- the multiline string, the heredoc below, and the loop over the associative array. On balance, that felt like not enough to outweigh having so much command line marshalling in Python.

Maybe there is a great Git API in Python that would avoid most of the commands, but I'm totally out of my depth there.

If this is important, I'll work on figuring out how to do it, but it'll take me some work.

**Review by chandlerc — COMMENTED** [review]
Everything tested and working. You can see it mirroring things here:
https://github.com/carbon-language/vim-carbon-lang/commits/trunk

I think this is in good working order now.

**Review by jonmeow — APPROVED** [review]
Not merging since you should probably watch this after it goes in.

**Pull request merged**
